### ISSUE TITLE
SignedXml is not a constructor (Angular/Browser)

### ISSUE THREAD

**Issue by gabucito** [PeculiarVentures/xadesjs#95]
Hello, I was trying to test the library inside an angular application (in the browser), but I have ran into a few problems.

The first problem was
`ReferenceError: regeneratorRuntime is not defined`
Although I am not targetting ES5,  I solved the problem by installing babel-polyfill
`npm install babel-polyfill --D`

However, now I get 
`xadesjs__WEBPACK_IMPORTED_MODULE_2__.SignedXml is not a constructor` 
which has got me stuck. The whole code is as follows:

```
import { Component } from '@angular/core';
import { SignedXml }  from 'xadesjs';

@Component({
  selector: 'app-xadesjs',
  templateUrl: './xadesjs.component.html',
  styleUrls: ['./xadesjs.component.scss']
})
export class XadesjsComponent {
  cryptoKey: CryptoKey;
  test() {
    let signedXml = new xadesjs.SignedXml();
  }
}
```

**Comment by microshine** [comment]
Looks like your compiler takes ES5 version form `xadesjs`. But module supports 3 compiled versions
`ES5`, `ES2105` and bundle ES5 file for browser. Use `module` instead of `main`

### package.json
```jsonc
{
  "name": "xadesjs",
  "version": "2.0.16",
  "description": "A pure Typescript/Javascript implementation of XAdES based on XMLDSIGjs.",
  "main": "build/index.js",
  "types": "build/types/index.d.ts",
  "module": "build/index.es.js",
  "browser": "build/xades.js",
  "browserMin": "build/xades.min.js",
  // ...
}
```

![image](https://user-images.githubusercontent.com/7333669/76603505-23aa2980-651e-11ea-974e-9633c38b71a0.png)

**Comment by microshine** [comment]
Just found some problems in your script.

```js
import { SignedXml }  from 'xadesjs';
let signedXml = new xadesjs.SignedXml();
```
 use

```js
import { SignedXml }  from 'xadesjs';
let signedXml = new SignedXml();
```
or
```ts
import * as xadesjs  from 'xadesjs';
let signedXml = new xadesjs.SignedXml();
```

**Comment by gabucito** [comment]
Sorry, I actually do have
``` typescript
import { SignedXml }  from 'xadesjs';
let signedXml = new SignedXml();
```
in my code, but I was messed up when trying to clean up the code to post in here.

I also did try
``` typescript
import * as xadesjs  from 'xadesjs';
let signedXml = new xadesjs.SignedXml();
```
but still get the same error.

**Comment by gabucito** [comment]
Sorry to trouble you again, I did fix the script but I did not understand the part where you said that I need to use `module` instead of `main`.

I took a look at the package.json from the imported xadesjs and this is what it looks like.
```json
{
  "_from": "xadesjs",
  "_id": "xadesjs@2.0.16",
  "_inBundle": false,
  "_integrity": "sha512-N5yqRvcTwQUch9gDaaafQRT/uCFimHxyO+KtY4wH/Jkqm6osGqxIoaawsZJ/JAQyfs8XiM48Z+pOp5OYh7qVFw==",
  "_location": "/xadesjs",
  "_phantomChildren": {},
  "_requested": {
    "type": "tag",
    "registry": true,
    "raw": "xadesjs",
    "name": "xadesjs",
    "escapedName": "xadesjs",
    "rawSpec": "",
    "saveSpec": null,
    "fetchSpec": "latest"
  },
  "_requiredBy": [
    "#USER",
    "/"
  ],
  "_resolved": "https://registry.npmjs.org/xadesjs/-/xadesjs-2.0.16.tgz",
  "_shasum": "58fe12379257323101dec3d2eb3af7c204f91de9",
  "_spec": "xadesjs",
  "_where": "/home/gabu/System/tokki-system",
  "author": {
    "name": "Peculiar Ventures, Inc."
  },
  "browser": "build/xades.js",
  "browserMin": "build/xades.min.js",
  "bugs": {
    "url": "https://github.com/PeculiarVentures/xadesjs/issues"
  },
  "bundleDependencies": false,
  "contributors": [
    {
      "name": "Miroshin Stepan",
      "email": "microshine@mail.ru"
    }
  ],
  "dependencies": {
    "xml-core": "^1.0.20",
    "xmldsigjs": "^2.0.27"
  },
  "deprecated": false,
  "description": "A pure Typescript/Javascript implementation of XAdES based on XMLDSIGjs.",
  "devDependencies": {
    "@babel/core": "^7.7.7",
    "@babel/plugin-proposal-class-properties": "^7.7.4",
    "@babel/plugin-proposal-object-rest-spread": "^7.7.7",
    "@babel/preset-env": "^7.7.7",
    "@peculiar/webcrypto": "^1.0.22",
    "@rollup/plugin-node-resolve": "^6.0.0",
    "@types/dateformat": "^3.0.1",
    "@types/mocha": "^5.2.7",
    "@types/node": "^12.12.21",
    "coveralls": "^3.0.9",
    "mocha": "^6.2.2",
    "nyc": "^14.1.1",
    "rollup": "^1.27.13",
    "rollup-plugin-babel": "^4.3.3",
    "rollup-plugin-commonjs": "^10.1.0",
    "rollup-plugin-terser": "^5.1.3",
    "rollup-plugin-typescript2": "^0.25.3",
    "ts-node": "^8.5.4",
    "tslib": "^1.10.0",
    "tslint": "^5.20.1",
    "typescript": "^3.7.3"
  },
  "homepage": "https://github.com/PeculiarVentures/xadesjs#readme",
  "keywords": [
    "xml",
    "xmldsig",
    "xades",
    "webcrypto",
    "signature",
    "rsa",
    "ec"
  ],
  "license": "MIT",
  "main": "build/index.js",
  "module": "build/index.es.js",
  "name": "xadesjs",
  "nyc": {
    "extension": [
      ".ts",
      ".tsx"
    ],
    "include": [
      "src/**/*.ts"
    ],
    "exclude": [
      "**/*.d.ts"
    ],
    "reporter": [
      "text-summary",
      "html"
    ]
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/PeculiarVentures/xadesjs.git"
  },
  "scripts": {
    "build": "npm run build:module && npm run build:types",
    "build:module": "rollup -c",
    "build:types": "tsc -p tsconfig.types.json",
    "clear": "rimraf build/*",
    "coverage": "nyc npm test",
    "coveralls": "nyc report --reporter=text-lcov | coveralls",
    "lint": "tslint -p .",
    "lint:fix": "tslint --fix -p .",
    "postpub": "git push && git push --tags origin master",
    "postpub:next": "git push",
    "prepare": "npm run build",
    "prepub": "npm run lint && npm run build",
    "prepub:next": "npm run lint && npm run build",
    "pub": "npm version patch && npm publish",
    "pub:next": "npm version prerelease --preid=next && npm publish --tag next",
    "rebuild": "npm run clear && npm run build",
    "sync": "git ac && git pull --rebase && git push",
    "test": "npm run lint && mocha"
  },
  "types": "build/types/index.d.ts",
  "version": "2.0.16"
}
```
If I am understanding this correctly, it is already using `module` in the `build` `script`.(?)

**Comment by microshine** [comment]
https://github.com/rollup/rollup-plugin-node-resolve#usage
![image](https://user-images.githubusercontent.com/7333669/77051535-7f702900-69dc-11ea-90ad-8276ea47b3b8.png)

![image](https://user-images.githubusercontent.com/7333669/77051437-551e6b80-69dc-11ea-8bad-9f8cf30b923c.png)

**Comment by WrongSideOfHeaven** [comment]
Hello @gabucito , my problem is very similar. 
```TS
import {Component, OnInit} from '@angular/core';
import * as xadesjs  from 'xadesjs';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent implements OnInit {

  ngOnInit() {
    const signedXml = new xadesjs.SignedXml();
  }
}
```
Do you know how to deal with that situation ?
![Screenshot_3](https://user-images.githubusercontent.com/26686769/146237924-d6ed53e4-0020-4269-afff-37b231545a06.png)

**Comment by microshine** [comment]
Looks like the problem is in the `browser` field. Some bundlers use that filed for bundler. But that module doesn't export any classes :(

https://github.com/PeculiarVentures/xadesjs/blob/master/package.json#L8

It's similar to [this](https://github.com/PeculiarVentures/x509/issues/13) issue

I think we need to republish the module and use that same solution

**Comment by eidins** [comment]
Same issue with ReactJS

```
xadesjs__WEBPACK_IMPORTED_MODULE_4__.SignedXml is not a constructor
```

https://github.com/eidins/web-signer/blob/45a1269406a78a45e4a806d4bb31801109ab87d7/package.json

https://github.com/eidins/web-signer/blob/45a1269406a78a45e4a806d4bb31801109ab87d7/src/App.js#L5

**Comment by microshine** [comment]
I've published `xadesjs@2.3.0`. It uses `unpkg` instead of `browser` option and should fix the current issue.

Please try the latest version

**Comment by eidins** [comment]
Version 2.3.0 works for me

https://github.com/eidins/web-signer/blob/2b2670b578f7284bda57817a84c357c9b42812ee/src/App.js#L5
